src/HotChocolate/Fusion-vnext/test/Fusion.Execution.Tests/FusionTestBase.cs

```diff
@@ -14,14 +14,21 @@ protected static CompositeSchema CreateCompositeSchema()
         return CompositeSchemaBuilder.Create(compositeSchemaDoc);
     }
 
-    protected static RootPlanNode PlanOperationAsync(CompositeSchema compositeSchema, string operation)
+    protected static RootPlanNode PlanOperation(DocumentNode request, CompositeSchema compositeSchema)
     {
-        var doc = Utf8GraphQLParser.Parse(operation);
-
         var rewriter = new InlineFragmentOperationRewriter(compositeSchema);
-        var rewritten = rewriter.RewriteDocument(doc, null);
+        var rewritten = rewriter.RewriteDocument(request, null);
 
         var planner = new OperationPlanner(compositeSchema);
         return planner.CreatePlan(rewritten, null);
     }
+
+    protected static async Task MatchSnapshotAsync(DocumentNode request,RootPlanNode plan)
```

Comment:
should we add the original request document to the RootPlanNode and then integrate it into the yaml?

Reply:
This would simplify the snapshot

Reply:
I think having the rewritten / normalized document produced by the inline fragment rewriter would make more sense, but yes 👍

Reply:
OK ... I will do a quick PR